modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -126,6 +130,59 @@ public static MetadataType getMetadataType(Class<?> type, ClassTypeLoader typeLo
   }
 
   /**
+   * Transforms a {@link MetadataType} and generates the correspondent {@link DataType}
+   *
+   * @param metadataType to introspect a create a {@link DataType} from it.
+   * @return a {@link DataType} based on the given {@link MetadataType}
+   */
+  public static DataType toDataType(MetadataType metadataType) {
+    Class<?> type = getType(metadataType);
+    Reference<DataType> dataType = new Reference<>();
+
+    metadataType.accept(new MetadataTypeVisitor() {
+
+      @Override
+      protected void defaultVisit(MetadataType metadataType) {
+        dataType.set(DataType.fromType(type));
+      }
+
+      @Override
+      public void visitArrayType(ArrayType arrayType) {
+        Class itemClass = getType(arrayType.getType());
+        if (Collection.class.isAssignableFrom(type)) {
+          dataType.set(DataType.builder()
+              .collectionType((Class<? extends Collection>) type)
+              .itemType(itemClass)
+              .build());
+        } else if (Iterator.class.isAssignableFrom(type)) {
+          dataType.set(DataType.builder()
+              .streamType((Class<? extends Iterator>) type)
+              .itemType(itemClass)
+              .build());
+        } else {
+          defaultVisit(arrayType);
+        }
+      }
+
+      @Override
+      public void visitObject(ObjectType objectType) {
+        if (objectType.isOpen() && Map.class.isAssignableFrom(type)) {
```

Comment:
if the type is a Map, you shouldn't care about it being open or not. It's a map and DW should treat it as such